Settings.StyleCop

```diff
@@ -27,6 +40,18 @@
             <BooleanProperty Name="Enabled">False</BooleanProperty>
           </RuleSettings>
         </Rule>
+        <Rule Name="UseBuiltInTypeAlias">
+          <!-- it's OK to use 'String' for API calls -->
```

Comment:
(!) We already fixed tons of such issues. It is not good use different styles in different files.
So I'd not turn out the rule now.

Reply:
There are 4676 violations of this rule, and most for the occurrences are member access or method invocations on `String`. It's good to always use `string` for simple type references in declaring a variable or so. When accessing a member or calling a method on `System.String`, I don't think using `String` is a problem or will cause any readability issue. It just not worthwhile to fix them all.

Reply:
We can probably defer this for now.  Although I think we could fix these with a regex in vscode to search/replace.

Reply:
https://github.com/PowerShell/PowerShell/pull/8520

Reply:
@SteveL-MSFT Be careful that we do want to use the full name in the places that we are explicitly dealing with different primitive types, for example, 
https://github.com/PowerShell/PowerShell/blob/8d27eb44b10bf48d8a47e50dd75d23b06c534270/src/System.Management.Automation/CoreCLR/CorePsStub.cs#L312-L329

Those cases are in the interpreter code, COM related code, serialization related code, type conversion related code, type resolver related code, and possibly other places.
And you need to be careful about the enum members of `TypeCode`, which has the same name as the full names of those primitive types. (and similar cases where the full names of those types are used as strings)

In my opinion, fixing those occurrences doesn't bring much benefit, but is expensive to do and to review.

Reply:
@daxian-dbw it was a very targeted search replace so your examples are not affected

Reply:
Some work had been done to correct many instances that violated this rule since this conversation started. So I will keep this rule enabled.